provider/kubernetes/kubernetes.go

```diff
@@ -287,8 +286,13 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 							}
 
 							for _, subset := range endpoints.Subsets {
+								endpointPort := endpointPortNumber(port, subset.Ports)
+								if endpointPort == 0 {
```

Comment:
I don't think this is intuitive. I would lean towards a proper error being returned, and a message logged

Reply:
Logging wouldn't make too much sense here IMHO since it's sometimes totally legitimate to not find matching endpoints: If you have a multi-port service with ports A and B, there will also be endpoints ports A and B. So service port A will never match endpoints point B, and service port B will never match endpoints point A. I think it'd be quite noisy if we logged every such combo, even at DEBUG level.